docs/faq/DesignDecisions.md

```diff
@@ -0,0 +1,75 @@
+# Redux FAQ: Design Decisions
+
+## Table of Contents
+
+- [Why doesn't Redux pass the state and action to subscribers?](#does-not-pass-state-action-to-subscribers) 
+- [Why doesn't Redux support using classes for actions and reducers?](#does-not-support-classes) 
+- [Why does the middleware signature use currying?](#why-currying)
+- [Why does applyMiddleware use a closure for dispatch?](#closure-dispatch)
+- [Can you please change combineReducers to support nested state trees?](#combineReducers-limitations)
+- [Why doesn't mapDispatchToProps allow use of return values from getState() or mapStateToProps()?](#no-asynch-in-mapDispatchToProps)
+
+
+## Design Decisions
+
+<a id="does-not-pass-state-action-to-subscribers"></a>
+### Why doesn't Redux pass the state and action to subscribers?
+Subscribers are intended to respond to the state value itself, not the action. Updates to the state are not always processed synchronously, because libraries can change Redux to process actions in batches to optimize performance and avoid repeated re-rendering. The intended guarantee is that Redux eventually calls all subscribers with the most recent state, but not that it always calls each subscriber for each action. The store state is available in the subscriber simply by calling store.getState(). The action cannot be made available in the subsribers without breaking the way that actions are batched. 
+
+A potential use-case for using the action inside a subscriber -- which is an unsupported feature -- is to ensure that a component only re-renders after certain kinds of actions. Re-rendering should instead be controlled instead through:
+1.) the [shouldComponentUpdate](https://facebook.github.io/react/docs/react-component.html#shouldcomponentupdate) lifecycle method
+2.) the [virtual DOM equality check (vDOMEq)](https://facebook.github.io/react/docs/optimizing-performance.html#avoid-reconciliation)
+3.) [React.PureComponent](https://facebook.github.io/react/docs/optimizing-performance.html#examples)
+4.) Using React-Redux: use [mapStateToProps](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) to subscribe components to only the parts of the store that they need.
+
+<a id="does-not-support-classes"></a>
+### Why doesn't Redux support using classes for actions and reducers?
+The pattern of using functions, called action creators, to return action objects may seem counterintuitive to programmers with a lot of Object Oriented Programming experience, who would see this is a strong use-case for Classes and instances. Class instances for action objects and reducers are not supported because class instances make serialization and deserialization tricky. Deserialization methods like JSON.parse(string) will return a plain old Javascript object rather than class instances. 
```

Comment:
General request: add backticks for inline code formatting to all mentions of function names and stuff like `JSON.parse()`.